src/test/java/org/fogbowcloud/manager/core/AaControllerTest.java

```diff
@@ -48,112 +48,174 @@ public void setUp() {
     	
     }
     
+    //test case: Check if authenticate method throws no exception when the federation token is valid.
     @Test
     public void testAuthenticate() throws UnauthenticatedUserException {
+    	//set up
```

Comment:
In addition to that, I'd make sure that the same string passed as arg in this.asController.authenticate is passed to the federationIdentityPlugin (so, add a verify to this test case)

Reply:
Adding verify is really a good way to improve this type of test, thank you for the observation